### PR TITLE
Dio loggy to null safety

### PR DESCRIPTION
**Pull request by lukaknezic**

### CODE REVIEW

File: flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart [hunk 1]
```diff
@@ -164,9 +153,9 @@ class LoggyDioInterceptor extends Interceptor with DioLoggy {
 
   void _commit(LogLevel level) {
     if (level.priority >= LogLevel.error.priority) {
-      final _valueError = _value.toString();
-      final _errorTitle = _valueError.substring(0, _valueError.indexOf('\n'));
-      final _errorBody = _valueError.substring(_errorTitle.length);
+      final String _valueError = _value.toString();
+      final String _errorTitle = _valueError.substring(0, _valueError.indexOf('\n'));
+      final String _errorBody = _valueError.substring(_errorTitle.length);
```
Comment (Truba) [inline flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart:158]:
```suggestion
      final _valueError = _value.toString();
      final _errorTitle = _valueError.substring(0, _valueError.indexOf('\n'));
      final _errorBody = _valueError.substring(_errorTitle.length);
```
This was ok, as it was 😄

File: flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart [hunk 2]
```diff
@@ -54,48 +52,41 @@ class LoggyDioInterceptor extends Interceptor with DioLoggy {
       _prettyPrintObject(options.extra, header: 'Extras');
     }
     if (requestBody && options.method != 'GET') {
-      final dynamic data = options.data;
+      final Object? data = options.data;
       if (data != null) {
         if (data is FormData) {
```
Comment (Truba) [inline flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart:57]: Early return please 😄

File: flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart [hunk 3]
```diff
@@ -54,48 +52,41 @@ class LoggyDioInterceptor extends Interceptor with DioLoggy {
...
 
     _commit(requestLevel ?? LogLevel.info);
-    return options;
   }
 
   @override
-  Future onError(DioError err) async {
+  void onError(DioError err, ErrorInterceptorHandler handler) async {
     if (error) {
       if (err.type == DioErrorType.response) {
```
Comment (Truba) [inline flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart:72]: Early return please 😄

File: flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart [hunk 4]
```diff
@@ -54,48 +52,41 @@ class LoggyDioInterceptor extends Interceptor with DioLoggy {
...
         logPrint(
-            '<<< DioError │ ${err.request?.method} │ ${err.response?.statusCode} ${err.response?.statusMessage} │ ${err.response?.request.uri.toString()}');
+            '<<< DioError │ ${err.requestOptions.method} │ ${err.response?.statusCode} ${err.response?.statusMessage} │ ${err.response?.requestOptions.uri.toString()}');
         if (err.response != null && err.response!.data != null) {
-          _prettyPrintObject(err.response!.data,
-              header: 'DioError │ ${err.type}');
+          _prettyPrintObject(err.response!.data, header: 'DioError │ ${err.type}');
```
Comment (Truba) [inline flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart:76]: You should always avoid force casting, dart should have a way to safely unwrap an optional 😄

File: flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart (hunk 1 above)
Comment (lukaknezic) [reply flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart:158]: We have `pedantic` as analysis options in here, and analysis is not happy with previous solution 😄

**Review by Truba — APPROVED** [review]

**Pull request merged**